client/src/components/header/header.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Created by einavcarmon on 15/06/2017.
+ */
+import React, { Component } from 'react';
+import { Button } from 'react-bootstrap';
+import Strings from '../../strings';
+import Login from '../login'
+import { logout } from '../../api.js'
+
+import './header.css'
+
+export default class Header extends Component {
+
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            user: props.user
+        }
+    }
+
+    logout() {
+        logout();
+    }
+
+    render () {
+        return (
+            <div>
+                <div className="App-header">
+                    <img src={ process.env.PUBLIC_URL + '/img/logo_alpha.png'} className="App-logo" alt="logo" />
+                </div>
+                { !this.state.user && (
+                    <div className="login-container"> {Strings.header.loginHeader}
+                        <a className="inline" href="#" onClick={() => this.setState({showLogin: true})} >
+                            <Button bsStyle="primary"> { Strings.header.loginAction } </Button>
+                        </a>
+                    </div>
+                )}
+
+                { this.state.user && (
+                    <div className="login-container">
+                        <p className="inline logged-in-header">{Strings.header.welcome}, {this.state.user.firstName} !</p> <span>
+                        <a className="inline logout-header" href="#" onClick={() => this.logout()}>{Strings.header.logout}</a></span>
+                    </div>
+                )}
+
+                <Login
+                    onHide={ () => { this.setState({showLogin: false}) } }
+                    show={this.state.showLogin}
+                    onLogin={ (user) => {
+                        this.setState({
```

Comment:
to my understanding it's usually a bad sign if you have the same piece of information (in this case `user`) managed both in the props (by the containing component) and the state.
 
Specifically in this implementation this seems to actually cause a bug when a logged in user refreshes the page - the header won't show that the user is logged in. This is because the way I implemented user fetching in a logged-in session, a logged-in user will be retrieved *after* the header component is constructed, so it would change on the the header's props but not on its state...